plugins/pdi-pur-plugin/src/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2378,7 +2378,9 @@ private void updateSharedObjectCache( final RepositoryElementInterface element,
     // if there's a match on id, replace the element
     boolean found = false;
     for ( int i = 0; i < origSharedObjects.size(); i++ ) {
-      if ( ( (RepositoryElementInterface) origSharedObjects.get( i ) ).getObjectId().equals( idToFind ) ) {
+      RepositoryElementInterface repositoryElementInterface = (RepositoryElementInterface) origSharedObjects.get( i );
+      ObjectId objectId = repositoryElementInterface.getObjectId();
```

Comment:
i don't know, probably. That wasn't what was causing the issue i was dealing with, it was the object id. want me to add another check?